lib/msal-node/src/cache/TokenCache.ts

```diff
@@ -97,6 +97,25 @@ export class TokenCache implements ISerializableTokenCache, ITokenCache {
         return this.storage.getCache();
     }
 
+    /**
+     * API that restores cache context from persistence
+     */
+    async restoreCacheContext(): Promise<void> {
+        this.logger.trace("refreshCacheContext called");
+        if (this.persistence && StringUtils.isEmpty(this.cacheSnapshot)) {
```

Comment:
`CacheSnapShot` not being empty does not signify it being current, does it? If that is the case, we might as well just check for persistence. Else we may have a cache hit but will not have accuracy.

Reply:
Ok, I think I see your point. My intention was to avoid running `beforeCacheAccess` unnecessarily since it will have `some` performance impact, but in a distributed scenario we cannot guarantee cacheSnapshot accuracy. In that case, I suppose it we should not do this check and accept the perf hit :/

Reply:
Sorry for the late reply. I've thought more about this -you're right, we cannot make this assumption i.e. in-memo cache needs to be synced with disk cache before and after each `acquireToken` call to ensure consistency (e.g. app1 and app2 shares cache, later app2 logs out the user, removes tokens from the shared cache, but app1 still has tokens in the in-memory cache. expectation would be that app1 brings its in-memo cache to sync before making another token request). As such, I've removed the check.
 
From what I see, performance hit seems to be negligible with local cache -I'll measure again with a remote location and update just in case. 

One potential inefficiency is due to usage: if a dev is calling `getAccount*` right before `acquireToken*`, `beforeCacheAccess`/`afterCacheAccess` pair will be called twice in a row. Any thoughts about this? @sameerag @bgavrilMS

Reply:
following up on this -a remote cache service like Azure Redis seem to incur about a 30ms latency on average.

Reply:
This is probably too high per token fetch. Let us discuss once, if we have a better solution.